sumologic/resource_sumologic_generic_polling_source.go

```diff
@@ -590,6 +597,7 @@ func getPollingPathSettings(d *schema.ResourceData) (PollingPath, error) {
 					LimitToNamespaces = append(LimitToNamespaces, v.(string))
 				}
 			}
+			pathSettings.UseVersionedApi = false
```

Comment:
does this need to be set to false for all path except `S3BucketPathExpression`? just wondering why other path properties like `TagFilters` that also only applies to subset of path types are not set to a bogus value in path types that don't apply

Reply:
other properties don't have any default value assigned to them as we have for `use_versioned_api` parameter, that is why I have explicitly set it to false for other paths.

Reply:
why is that? tag filters is also an optional field on cloudwatch source for example

Reply:
even though `tag_filters` is optional, it does not have a default value assigned to it. We can also try removing the default value for `use_versioned_api` and only set it if required.

Reply:
I think that makes more sense because we don't want to send `false` to other source types that don't recognize this property. just need to make sure if user doesn't specify it for S3 source it will still default to `false` on the backend

Reply:
also, the default should be true